types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -570,10 +570,12 @@ void wlr_xdg_popup_destroy(struct wlr_xdg_surface *surface) {
 
 static void xdg_popup_get_position(struct wlr_xdg_popup *popup,
 		double *popup_sx, double *popup_sy) {
-	struct wlr_xdg_surface *parent =
-		wlr_xdg_surface_from_wlr_surface(popup->parent);
-	struct wlr_box parent_geo;
-	wlr_xdg_surface_get_geometry(parent, &parent_geo);
+	struct wlr_box parent_geo = {0};
+	if (wlr_surface_is_xdg_surface(popup->parent)) {
+		struct wlr_xdg_surface *parent =
+			wlr_xdg_surface_from_wlr_surface(popup->parent);
+		wlr_xdg_surface_get_geometry(parent, &parent_geo);
+	}
```

Comment:
Perhaps I've misunderstood the issue, but don't we need to use the parent coordinates if the parent is a layer surface as well? Currently I don't see how this PR actually changes the result.

Reply:
No, layer surfaces don't have "geometry" in the same sense as xdg surfaces do. For example, let's say we have this:
- A
  - Popup 1
    - Popup 2
 If A is an xdg surface, then popup 2 would have its coordinates + popup 1 coordinates + A coordinates.
 If A is an layer surface, then popup 2 would have only its coordinates. Coordinates of A don't matter in this case, as layer surfaces don't have a position. Howerer, popup 1 *has* a position, and it must be added to the position of popup 2, which is what this PR does.

Reply:
Actually I think this explanation is incorrect. Doesn't matter now anyway.